cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/DBSource.java

```diff
@@ -124,6 +131,48 @@ private String getJDBCPluginId() {
     return String.format("%s.%s.%s", "source", dbSourceConfig.jdbcPluginType, dbSourceConfig.jdbcPluginName);
   }
 
+  private void ensureValidConnection(Class<? extends Driver> jdbcDriverClass) {
+    try {
+      ensureJDBCDriverIsAvailable(jdbcDriverClass);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("could not ensure JDBCDriver is available", e);
+    }
+    try (Connection connection = createConnection()
+    ) {
+      assert(connection.isValid(0));
+    } catch (SQLException e) {
+      throw new IllegalArgumentException("SQL Exception thrown when trying to connect to driver", e);
+    } finally {
+      DBUtils.cleanup(jdbcDriverClass);
```

Comment:
Since now the `configure` method also instantiate SQL connection, it needs to be called in a `finally` block of the `configure` method as well, as destroy won't be called at configure time.